android/src/main/kotlin/com/wire/kalium/KaliumApplication.kt

```diff
@@ -1,5 +1,28 @@
 package com.wire.kalium
 
 import android.app.Application
+import androidx.work.*
+import com.wire.kalium.logic.CoreLogic
+import com.wire.kalium.logic.sync.WrapperWorkerFactory
+import java.io.File
 
-class KaliumApplication: Application()
+class KaliumApplication: Application(), Configuration.Provider {
+
+    lateinit var coreLogic: CoreLogic
+
+    override fun onCreate() {
+        super.onCreate()
+
+        val rootProteusDir = File(this.filesDir, "proteus")
+        coreLogic = CoreLogic(applicationContext, "kalium", rootProteusDir.absolutePath)
+    }
+
+    override fun getWorkManagerConfiguration(): Configuration {
+        val myWorkerFactory = WrapperWorkerFactory(coreLogic)
+
+        return Configuration.Builder()
+            .setWorkerFactory(myWorkerFactory)
+            .build()
+    }
```

Comment:
I wonder if we could add this implementation to `CoreLogic/androidMain` so it would be easier to use plug-in into an Android app outside the box.

Not needed now, but something to keep in mind.